sktm/patchwork.py

```diff
@@ -418,6 +418,13 @@ def _get_mbox_url_sfx(self):
 
         return "mbox"
 
+    def _get_project_id(self, project_name):
+        """
+        Get project ID based on its name. Child classes need to implement
+        this method.
+        """
+        raise NotImplementedError
```

Comment:
Can I ask you to write full documentation for this method? I think it's even more important to document than the implementations, as it defines the interface.